### PR TITLE
Browser: Fix placeholders part two

### PR DESCRIPTION
**Pull request by varjolintu**
[NOTE]: # ( Describe your changes in detail. Explain large or complex code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX". )
[NOTE]: # ( If you used Generative AI to write the majority of your code, you must state this. )
Continues from https://github.com/keepassxreboot/keepassxc/pull/13580. If a placeholder was wrapped inside another brackets, for example `{{PASSWORD}}`, it was parsed as an unknown type, but was still resolved to an actual value later. 

This scenario happened when trying to store credentials via browser extension. If `placeholdeType()` returns an unknown placeholder, it allows saving a reference to an entry.

Added a `maxDepth` to `placeholderType()` check that verifies possible placeholder types inside extra brackets. If max depth is exceeded, the placeholder is returned as non-resolved raw value.

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and include helpful comments. )
Automatic tests added. Any edge cases must be added if identified.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change that fixes an issue)

### CODE REVIEW

File: tests/TestEntry.cpp [hunk 1]
```diff
@@ -952,5 +959,11 @@ void TestEntry::testContainsPlaceholder()
     QVERIFY(EntryPlaceholders::containsPlaceholder("{NOTES}"));
     QVERIFY(EntryPlaceholders::containsPlaceholder("inthe{NOTES}middle"));
     QVERIFY(EntryPlaceholders::containsPlaceholder("{TOTP}"));
+    QVERIFY(EntryPlaceholders::containsPlaceholder("{{TOTP}}"));
+    QVERIFY(EntryPlaceholders::containsPlaceholder("after{{TOTP}}"));
     QVERIFY(EntryPlaceholders::containsPlaceholder("test\\{TOTP\\}"));
+
+    // Max depth (10), and max depth exceeded
+    QVERIFY(EntryPlaceholders::containsPlaceholder("{{{{{{{{{{TOTP}}}}}}}}}}"));
+    QVERIFY(!EntryPlaceholders::containsPlaceholder("{{{{{{{{{{{TOTP}}}}}}}}}}}"));
```
Comment (droidmonkey) [inline tests/TestEntry.cpp]: Actually using the value of the max depth and constructing the test around that would ensure any changes to the value in the future are covered. So prepending and appending the max depth number of braces for the first test, then increment by 1 and do that again.

**Review by droidmonkey — APPROVED** [review]
